src/Development/IDE/Core/OfInterest.hs

```diff
@@ -80,8 +80,8 @@ modifyFilesOfInterest state f = do
     OfInterestVar var <- getIdeGlobalState state
     files <- modifyVar var $ pure . dupe . f
     logDebug (ideLogger state) $ "Set files of interest to: " <> T.pack (show $ HashSet.toList files)
-    let das = map (\nfp -> mkDelayedAction "OfInterest" Debug (use GetSpanInfo nfp)) (HashSet.toList files)
-    shakeRestart state das
+    -- let das = map (\nfp -> mkDelayedAction "OfInterest" Debug (use GetSpanInfo nfp)) (HashSet.toList files)
+    -- shakeRestart state das
```

Comment:
why comment this out?

Reply:
`modifyFilesOfInterest` is called on `didOpen` and `didClose`. The parent type-checking logic already handles re-typechecking the relevant files that could be effected, so typechecking all files of interest seems redundant. 

I've changed the `didClose` handler to also typecheck parents, so that should get rid of the need for this.